capitalize_decorator.rb

```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super().capitalize
```

Comment:
👍 Great implementation! I noticed that you're currently using `super().capitalize`, which is a good approach. However, I'd recommend an alternative to enhance your code. Consider replacing `super().capitalize` with `@nameable.capitalize` in your `CapitalizeDecorator` class's `correct_name` method.

Each approach has a different purpose:

- [ ] `super().capitalize`: This uses `super` to call the overridden method from the parent class (Decorator in this case). It first calls the `correct_name` method from the parent class and then capitalizes the result. This approach ensures that any changes made in the parent class's `correct_name` method are respected, and you're only modifying the result by capitalizing it.

- [ ] `@nameable.capitalize`: This directly operates on the `@nameable` instance variable without considering the behavior defined in the parent class's `correct_name` method. It bypasses any logic or functionality that might be implemented in the parent class and simply capitalizes the value stored in `@nameable`.

The goal with `@nameable.capitalize` is to have more direct control over the capitalization process without relying on the parent class's behavior. Moreover, by avoiding `super().capitalize`, you can decouple the `CapitalizeDecorator` from potential changes in the parent class. If you anticipate changes or variations in the parent class's `correct_name` method that should not affect the capitalization process in this decorator, using `@nameable.capitalize` ensures that your decorator remains independent. Keep up the good work! 👏📝💡